### PR TITLE
CDPT-2346 Add Entra mfa to the login page

### PR DESCRIPTION
**Pull request by EarthlingDavey**

### CODE REVIEW

File: public/app/mu-plugins/moj-auth/moj-auth.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        $jwt->roles = ['employee'];
+
+        // Set a JWT cookie.
+        $this->setJwt($jwt);
+
+        // Store the tokens.
+        $this->storeTokens($this->sub, $oauth_access_token, 'refresh');
+
+        // Ensure we're redirecting to a page on the same domain as our home_url.
+        if (empty($jwt->success_url) || !str_starts_with($jwt->success_url, home_url())) {
```
[the remark this answers is not in the document]
Comment (wilson1000) [reply public/app/mu-plugins/moj-auth/moj-auth.php:155]: Or, looking at the docs, pass a forward slash to the `home_url()` to ensure we are covered.
```
home_url() = https://www.justice.gov.uk
home_url('/') = https://www.justice.gov.uk/
```

**Review by wilson1000 — CHANGES_REQUESTED** [review]
Hi mate, everything looks awesome, just a note around host checking.

File: public/app/mu-plugins/moj-auth/moj-auth.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (EarthlingDavey) [reply public/app/mu-plugins/moj-auth/moj-auth.php:155]: Hey @wilson1000 good to hear it looks good.

I hear your point about `https://www.justice.gov.uk.spoofeddomain.com/`... 

In this case we have the benefit that we can guarantee that `$jwt` (and all of it's properties) have been set by our server. The `getJwt` function will return false if the cookie has an invalid signature.

I don't think there's any harm in belt & braces on this one, I'll improve the check to be more thourough.

**Comment by EarthlingDavey** [comment]
@wilson1000 , would you mind re-reviewing this?

**Review by wilson1000 — APPROVED** [review]
Looks good mate, thank you!

**Pull request merged**